File: programs/assignment3/ProgrammingAssignment3/ProgrammingAssignment3/Receptionist.cpp
// COSC 1320 Summer 2015
// Name: Adrian
// Programming Assignment 3
// This is my own work; I will not post

#pragma once
#include <string>
#include "Administrator.h"
#include "HospitalEmployee.h"
#include "Receptionist.h"
using namespace std;

Receptionist::~Receptionist(void)
{
}

Receptionist::Receptionist() : Administrator('r' , "none" , 0 , "Talking"){
	answering = 'N';
}
Receptionist::Receptionist(char newRole, string newName, int newID, string newDept, char busyState) : Administrator(newRole , newName , newID , newDept){
	answering = busyState;
}
char Receptionist::getAnswering(){
	return answering;
}
void Receptionist::setAnswering(char busyState){
	answering = busyState;
}
string Receptionist::toString(){
	string str = " ";
	str += answering;
	return (Administrator::toString() + str);
}
bool Receptionist::equals(HospitalEmployee obj){
	if(obj.getID() == getID() && obj.getName() == getName() && obj.getRole() == getRole()){
		return true;
	} else {
		return false;
	}
}

File: programs/assignment1/ProgrammingAssignment1/src/programmingassignment1/Receptionist.java
// COSC 1320 Summer 2015
// Name: Adrian
// Programming Assignment 1
// This is my own work; I will not post

package programmingassignment1;

/**
 * 
 * @author Adrian Davila
 */
public class Receptionist extends Employee{
    String department;
    private char answering;

    Receptionist(){
        department = "";
        answering = 'Y';
    }
    Receptionist(char newRole , String newName , int newID , String dept , char onCall){
        super(newRole , newName , newID);
        department = dept;
        answering = onCall;
        addEmployee((Employee)this);
    }

    public void deleteEmployee(){
        deleteEmployee(getName() , getRole());
    }

    // Mutators
    public void setDept(String newDept){
        department = newDept;
    }
    public void setworkStat(char workStat){
        answering = workStat;
    }
    // Accessors
    public String getDept(){
        return department;
    }
    public char getworkStat(){
        return answering;
    }

    @Override
    public String toString(){
        return (super.toString() + " "  + department + " " + answering);
    }
    @Override
    public String toString(boolean t){
        return (super.toString() + " Department: "  + department + " Answering: " + answering);
    }
    @Override
    public boolean equals(Employee obj){
        if(obj == null){
            return false;
        }else if(getClass() != obj.getClass()){
            return false;
        }else{
            Receptionist otherRec = (Receptionist)obj;
            return (getName().equals(otherRec.getName()) && getRole() == otherRec.getRole());
        }
    }
}
